tests/js/plugin/texthighlight-test.coffee

```diff
@@ -0,0 +1,78 @@
+assert = chai.assert
+sinon.assert.expose(assert, prefix: '')
+
+# In order to be able to create highlights,
+# the Annotator.TextHighlight class must exist.
+# This class is registered then the TextHighlights plugin
+# is initialized, so we will do that.
+th = new Annotator.Plugin.TextHighlights()
+th.pluginInit()
+
+describe 'Annotator.Plugin.TextHighlight', ->
+  sandbox = null
+  jqElement = null
+  scrollTarget = null
+
+  createTestHighlight = ->
+    anchor =
+      id: "test anchor"
+      annotation: "test annotation"
+      anchoring:
+        id: "test anchoring manager"
+        annotator:
+          id: "test annotator"
+          element:
+            delegate: sinon.spy()
+
+    new Annotator.TextHighlight anchor, "test page", "test range"
+
+  beforeEach ->
+    sandbox = sinon.sandbox.create()
+    sandbox.stub Annotator.TextHighlight, 'highlightRange',
+      (normedRange, cssClass) -> "test highlight span"
+
+    sandbox.stub Annotator.$.fn, "init", (selector, context) ->
+      jqElement =
+        selector: selector
+        data: sinon.spy()
+        scrollintoview: sinon.spy (options) ->
+          scrollTarget = this.selector
+          options?.complete?()
+
+  afterEach ->
+    sandbox.restore()
+    scrollTarget = null
+
+  describe "constructor", ->
+    it 'wraps a highlight span around the given range', ->
+      hl = createTestHighlight()
+      assert.calledWith Annotator.TextHighlight.highlightRange, "test range"
+
+    it 'stores the created highlight spans in _highlights', ->
```

Comment:
> The benefit is thinking in terms of concepts that exist in HTML specifications. If not an `Element`, we may have new `Node` subclasses associated with annotation / highlighting / selecting that we should try to discover and bring to those efforts.
> [...]
> From the perspective of a developer, it means API simplicity and consistency.

My main objection is that Anchors don't live in the DOM; they belong to the (more abstract) *document*, for which the DOM is only a window. (This distinction is important on all reader/viewer platforms where only a part of the document is rendered to the DOM at any single time.)
For the part of the document which actually appears in the DOM, the anchors manifest in the DOM as highlights. The part of the document that doesn't appear in the DOM is just not there; no content, no highlights.

That's why I think it would be feel wrong to force the Anchors into the DOM, in whatever shape and form. (DOM Elements, nodes, whatever.) I think they should be available via a different API.